src/sentry/interfaces/base.py

```diff
@@ -198,24 +199,89 @@ def get_api_meta(self, meta, is_public=False):
     def to_json(self):
         return prune_empty_keys(self._data)
 
-    def get_hash(self, platform=None, variant='system'):
-        return []
-
-    def get_hashes(self, platform=None):
-        system_hash = self.get_hash(platform, variant='system')
-        if not system_hash:
-            return {}
-
-        hashes = {'system': system_hash}
-
-        app_hash = self.get_hash(platform, variant='app')
-        if system_hash != app_hash and app_hash:
-            hashes['app'] = app_hash
+    def get_grouping_component(self, platform=None, variant=None):
```

Comment:
I kinda want to just make platform and variant mandatory. Right now variant is not mandatory because some code passes platform by position. The current explanation i made myself is that if no variant is passed then whatever is the default logic should be used. This is not too unreasonable for nested calls of interface code.